### PR TITLE
Architecture section revamp

### PR DESCRIPTION
**Pull request by premkumr**
- Added new Key Concepts
- Added detailed Query Layer - Parse/Analyze/Plan/Execute
- Added more sections in DocDB - Storage LSM/Packed rows/Data Model
- Renamed Sidebar titles
- Moved YB-Master & YB-Tserver to top level

### CODE REVIEW

**Review by aishwarya24 — COMMENTED** [review]
left a comment

File: docs/content/preview/architecture/yb-master.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+title: YB-Master service
+headerTitle: YB-Master service
+linkTitle: YB-Master
+description: Learn how the YB-Master service manages tablet metadata and coordinates cluster configuration changes.
+headcontent: Catalog information, tablet metadata and cluster coordination
```
Comment (ddhodge) [inline docs/content/preview/architecture/yb-master.md:6]:
```suggestion
headcontent: Catalog information, tablet metadata, and cluster coordination
```

File: docs/content/preview/architecture/yb-tserver.md [hunk 2]
```diff
@@ -1,76 +1,78 @@
 ---
 title: YB-TServer service
 headerTitle: YB-TServer service
-linkTitle: YB-TServer service
+linkTitle: YB-Tserver
```
Comment (ddhodge) [inline docs/content/preview/architecture/yb-tserver.md:4]:
```suggestion
linkTitle: YB-TServer
```

File: docs/content/preview/architecture/yb-tserver.md [hunk 3]
```diff
@@ -1,76 +1,78 @@
...
 description: Learn how the YB-TServer service stores and serves application data using tablets (also known as shards).
+headcontent: Tablet server maintains and manages tablets(a.k.a shards)
```
Comment (ddhodge) [inline docs/content/preview/architecture/yb-tserver.md:6]:
```suggestion
headcontent: Serving application data and managing tablets
```

**Review by ddhodge — COMMENTED** [review]
Some comments [not finished]

**Review by ddhodge — APPROVED** [review]

File: docs/content/preview/architecture/key-concepts.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+A cluster is a group of [nodes](#node) on which YugabyteDB is deployed. The table data is distributed across the various [nodes](#node) in the cluster. Typically used as [*Primary cluster*](#primary-cluster) and [*Read replica cluster*](#read-replica-cluster).
+
+{{<note>}}
+Sometimes the term *cluster* is used interchangeably with the term *universe*. However, the two are not always equivalent, as described in [Universe](#universe).
+{{</note>}}
+
+## DocDB
+
+DocDB is the underlying document storage engine of YugabyteDB and is built on top of a highly customized and optimized verison of [RocksDB](http://rocksdb.org/) {{<link "../docdb">}}
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:34]:
```suggestion
DocDB is the underlying document storage engine of YugabyteDB and is built on top of a highly customized and optimized verison of [RocksDB](http://rocksdb.org/). {{<link "../docdb">}}
```

File: docs/content/preview/architecture/key-concepts.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+## Fault domain
+
+A fault domain is a potential point of failure. Examples of fault domains would be nodes, racks, zones, or entire regions. {{<link "../../../explore/fault-tolerance/#fault-domains">}}
+## Master server
+
+The [YB-Master](../yb-master/) service is responsible for keeping system metadata, coordinating system-wide operations, such as creating, altering, and dropping tables, as well as initiating maintenance operations such as load balancing. {{<link "yb-master">}}
+
+{{<tip>}}
+The master server is also typically referred as just **master**
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:44]:
```suggestion
The master server is also typically referred as just **master**.
```

File: docs/content/preview/architecture/key-concepts.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+{{</tip>}}
+
+## MVCC
+
+MVCC stands for Multiversion Concurrency Control. It is a concurrency control method used by YugabyteDB to provide access to data in a way that allows concurrent queries and updates without causing conflicts. {{<link "../../transactions/transactions-overview/#mvcc-using-hybrid-time">}}
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:49]: broken link

File: docs/content/preview/architecture/key-concepts.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+## Namespace
+
+A namespace refers to a logical grouping or container for related database objects, such as tables, views, indexes, and other database constructs. Namespaces help organize and separate these objects, preventing naming conflicts and providing a way to control access and permissions.
+
+A namespace in YSQL is referred to as a database and is logically identical to a namespace in other RDBMS (such as PostgreSQL) and a namespace in YCQL is referred to as a keyspace and is logically identical to a keyspace in Apache Cassandra's CQL.
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:55]:
```suggestion
A namespace in YSQL is referred to as a database and is logically identical to a namespace in other RDBMS (such as PostgreSQL).

 A namespace in YCQL is referred to as a keyspace and is logically identical to a keyspace in Apache Cassandra's CQL.
```

File: docs/content/preview/architecture/key-concepts.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+## Node
+
+A node is a virtual machine, physical machine, or container on which YugabyteDB is deployed.
+
+## Leader balancing
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:61]: alphabetical order

File: docs/content/preview/architecture/key-concepts.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+
+YugabyteDB tries to keep the no.of leaders evenly distributed across the [nodes](#node) in a cluster to ensure an even distribution of load.
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:63]:
```suggestion
YugabyteDB tries to keep the number of leaders evenly distributed across the [nodes](#node) in a cluster to ensure an even distribution of load.
```

Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:63]: Maybe add an entry for "Leader"

File: docs/content/preview/architecture/key-concepts.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+## Primary cluster
+
+A primary cluster can perform both writes and reads. Replication between [nodes](#node) in a primary cluster is performed synchronously.
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:67]: As opposed to what? Why even make the distinction?

File: docs/content/preview/architecture/key-concepts.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+
+## Raft
+
+Raft stands for Replication for availability and fault tolerance. This is the algorithm that YugabyteDB uses for replication guaranteeing consistency. {{<link "../docdb-replication/replication/">}}
+
+## Read replica cluster
+
+Read replica clusters can perform only reads; writes sent to read replica clusters get automatically rerouted to the primary cluster of the [universe](#universe). These clusters enable reads in regions that are far away from the primary cluster with timeline-consistent data. This ensures low latency reads for geo-distributed applications.
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:75]:
```suggestion
Read replica clusters can perform only reads; writes sent to read replica clusters get automatically rerouted to the [primary cluster](#primary-cluster) of the [universe](#universe). These clusters enable reads in regions that are far away from the primary cluster with timeline-consistent data. This ensures low latency reads for geo-distributed applications.
```

File: docs/content/preview/architecture/key-concepts.md [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+Tablets are also referred as shards.
+{{</tip>}}
+
+## Tablet splitting
+
+When a tablet reaches a threshold size, it splits into 2 new [tablets](#tablet). This is a very quick operation. {{<link "../docdb-sharding/tablet-splitting">}}
+
+## TServer
+
+The [YB-TServer](../yb-tserver) service is responsible for maintaining and managing table data in the form of tablets, as well as dealing with all the queries. {{<link "yb-tserver">}}
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:107]: broken link

File: docs/content/preview/architecture/key-concepts.md [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+
+## Universe
+
+A YugabyteDB universe comprises one primary cluster and zero or more read replica clusters that collectively function as a resilient and scalable distributed database.
+
+{{<note>}}
+Sometimes the terms *universe* and *cluster* are used interchangeably. However, the two are not always equivalent, as a [Universe](#universe) can contain one or more [clusters](#cluster)
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:114]:
```suggestion
Sometimes the terms *universe* and *cluster* are used interchangeably. However, the two are not always equivalent, as a universe can contain one or more [clusters](#cluster).
```

File: docs/content/preview/architecture/key-concepts.md (hunk 10 above)
Comment (premkumr) [reply docs/content/preview/architecture/key-concepts.md:67]: As opposed to a read replica cluster. Will add that in

File: docs/content/preview/architecture/key-concepts.md [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+{{</note>}}
+
+## xCluster
+
+xCluster is the scheme of asynchronous replication between 2 [universes](#universe). Primary used for disaster recovery. YugabyteDB supports transactional xCluster {{<link "../docdb-replication/async-replication/">}}
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:119]:
```suggestion
xCluster is a type of deployment where data is replicated asynchronously between two [universes](#universe) - a primary and a standby. The standby can be used for disaster recovery. YugabyteDB supports transactional xCluster {{<link "../docdb-replication/async-replication/">}}.
```

File: docs/content/preview/architecture/key-concepts.md [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:121]: Add a docs link. We should also sort out the endless variations of primary and standby (active, master, follower, source, target, provider, consumer etc etc) that are used interchangeably everywhere across docs.

File: docs/content/preview/architecture/key-concepts.md [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+
+## YCQL
+
+Semi-relational SQL API that is best fit for internet-scale OLTP and HTAP apps needing massive write scalability as well as blazing-fast queries. It supports distributed transactions, strongly consistent secondary indexes and a native JSON column type. YCQL has its roots in the Cassandra Query Language. {{<link "../../api/ycql">}}
+
+## YQL
+
+The YugabyteDB Query Layer (YQL) is the primary layer that provides interfaces for applications to interact with using client drivers. This layer deals with the API specific aspects such as query/command compilation and the run-time (data type representations, built-in operations and more). {{<link "../query-layer">}}
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:129]:
```suggestion
The YugabyteDB Query Layer (YQL) is the primary layer that provides interfaces for applications to interact with using client drivers. This layer deals with the API-specific aspects such as query/command compilation and the run-time (data type representations, built-in operations, and more). {{<link "../query-layer">}}
```

File: docs/content/preview/architecture/key-concepts.md [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+
+## YSQL
+
+Fully-relational SQL API that is wire compatible with the SQL language in PostgreSQL. It is best fit for RDBMS workloads that need horizontal write scalability and global data distribution while also using relational modeling features such as JOINs, distributed transactions and referential integrity (such as foreign keys). Note that YSQL reuses the native query layer of the PostgreSQL open source project. {{<link "../../api/ysql">}}
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:133]:
```suggestion
Fully-relational SQL API that is wire compatible with the SQL language in PostgreSQL. It is best fit for RDBMS workloads that need horizontal write scalability and global data distribution while also using relational modeling features such as JOINs, distributed transactions, and referential integrity (such as foreign keys). Note that YSQL reuses the native query layer of the PostgreSQL open source project. {{<link "../../api/ysql">}}
```

File: docs/content/preview/architecture/key-concepts.md [hunk 18]
```diff
@@ -0,0 +1,137 @@
...
+
+## Zone
+
+Typically referred as Availability Zones or just AZ, zone is a datacenters or a group of colocated datacenters. Zone is the default fault domain in YugabyteDB.
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:137]:
```suggestion
Typically referred as Availability Zones or just AZ, a zone is a datacenter or a group of colocated datacenters. Zone is the default [fault domain](#fault-domain) in YugabyteDB.
```

File: docs/content/preview/architecture/key-concepts.md [hunk 19]
```diff
@@ -60,28 +64,28 @@ A namespace in YSQL is referred to as a database and is logically identical to a
...
 A node is a virtual machine, physical machine, or container on which YugabyteDB is deployed.
 
-## Leader balancing
-
-YugabyteDB tries to keep the number of leaders evenly distributed across the [nodes](#node) in a cluster to ensure an even distribution of load.
-
 ## Primary cluster
 
-A primary cluster can perform both writes and reads. Replication between [nodes](#node) in a primary cluster is performed synchronously.
+A primary cluster can perform both writes and reads unlike a [read replica cluster](#read-replica-cluster) that performs only reads. There is only one primary cluster in an [universe](#universe). Replication between [nodes](#node) in a primary cluster is performed synchronously.
```
Comment (ddhodge) [inline docs/content/preview/architecture/key-concepts.md:69]:
```suggestion
A primary cluster can perform both writes and reads, unlike a [read replica cluster](#read-replica-cluster), which can only serve reads. A [universe](#universe) can have only one primary cluster. Replication between [nodes](#node) in a primary cluster is performed synchronously.
```

File: docs/content/preview/architecture/query-layer/_index.md [hunk 20]
```diff
@@ -1,37 +1,95 @@
...
+
+1. Creating a Query Tree: The semantic analysis step creates a new data structure called the query tree, which represents the complete, semantically understood version of the query.
+
+The reason for separating the initial parsing and the semantic analysis is to allow certain types of queries (like transaction control commands) to be executed quickly without the need for a full semantic analysis. The query tree contains more detailed information about data types, functions, and expressions used in the query, making it easier for the system to execute the query correctly.
+
+### Analyzer
+
+The created query tree is then analyzed, rewritten and transformed based on any rules stored in the system catalog.
+
+Views are realized during this phase. Whenever a query against a view (i.e., a virtual table) is made, the original query is rewritten to a query that accesses the base tables given in the view definition instead.
```
[the remark this answers is not in the document]
Comment (premkumr) [reply docs/content/preview/architecture/query-layer/_index.md]: Postgres has virtual tables, essentially queries(like saved searches) against which queries can be written. The query is re-written to talk to the underlying tables. This is known as the realization of views.

**Pull request merged**